### PR TITLE
add torch2 compile to nlp pipelines

### PR DESCRIPTION
**Pull request by slin000111**
dialog_intent_prediction_pipeline
dialog_state_tracking_pipeline
document_grounded_dialog_generate_pipeline
document_grounded_dialog_rerank_pipeline
document_grounded_dialog_retrieval_pipeline
document_segmentation_pipeline
extractive_summarization_pipeline
feature_extraction_pipeline
fill_mask_pipeline
named_entity_recognition_pipeline
sentence_embedding_pipeline
siamese_uie_pipeline
table_question_answering_pipeline
text_generation_pipeline
text_ranking_pipeline
translation_evaluation_pipeline
user_satisfaction_estimation_pipeline
zero_shot_classification_pipeline

### CODE REVIEW

**Comment by wangxingjun778** [comment]
Ignore UT error: TranslationEvaluationPreprocessor: __init__() missing 1 required positional argument: 'eval_mode'

**Review by wangxingjun778 — APPROVED** [review]

File: modelscope/pipelines/nlp/document_segmentation_pipeline.py [hunk 1]
```diff
@@ -48,8 +48,14 @@ def __init__(
             preprocessor=preprocessor,
             config_file=config_file,
             device=device,
-            auto_collate=auto_collate)
-
+            auto_collate=auto_collate,
+            **kwargs)
+
+        kwargs = kwargs
```
[the remark this answers is not in the document]
Comment (slin000111) [reply modelscope/pipelines/nlp/document_segmentation_pipeline.py:54]: Thank you ! it has been removed in local file.

File: modelscope/pipelines/nlp/document_segmentation_pipeline.py [hunk 2]
```diff
@@ -48,8 +48,14 @@ def __init__(
...
+        if 'compile' in kwargs.keys():
```
[the remark this answers is not in the document]
Comment (slin000111) [reply modelscope/pipelines/nlp/document_segmentation_pipeline.py:55]: revised

File: modelscope/pipelines/nlp/document_segmentation_pipeline.py [hunk 3]
```diff
@@ -48,8 +48,14 @@ def __init__(
...
+            kwargs.pop('compile')
+        if 'compile_options' in kwargs.keys():
```
[the remark this answers is not in the document]
Comment (slin000111) [reply modelscope/pipelines/nlp/document_segmentation_pipeline.py:57]: revised

File: modelscope/pipelines/nlp/extractive_summarization_pipeline.py [hunk 4]
```diff
@@ -41,7 +41,14 @@ def __init__(
             preprocessor=preprocessor,
             config_file=config_file,
             device=device,
-            auto_collate=auto_collate)
+            auto_collate=auto_collate,
+            **kwargs)
+
+        kwargs = kwargs
```
[the remark this answers is not in the document]
Comment (slin000111) [reply modelscope/pipelines/nlp/extractive_summarization_pipeline.py:47]: removed

File: modelscope/pipelines/nlp/document_segmentation_pipeline.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wangxingjun778) [reply modelscope/pipelines/nlp/document_segmentation_pipeline.py:54]: ok, already merged to master

**Pull request merged**